# LAPD Crime Data Analysis (2020–2024)

This notebook analyzes crime incidents reported in Los Angeles between 2020 and 2024. I originally started this project using the Manitoba Rabies dataset, but after completing the initial exploration, it became clear that the dataset didn’t have enough depth or variation for meaningful data cleaning or analysis. 

The LAPD dataset provides a much stronger foundation for this course because it includes detailed information such as crime type, date and time, area, victim demographics, premises, weapon involvement, and geographic coordinates. It also contains real-world inconsistencies and missing values, which makes it more suitable for demonstrating proper data wrangling and exploratory analysis.

Because the full dataset is very large, I will be using a smaller random subset (`df_small`) for all analysis so the notebook runs efficiently while still reflecting realistic patterns in the data.


## 1. Initial Exploration

In [9]:
# Week 12: Initial dataset loading and inspection
# Purpose: Load the dataset and perform basic structural checks.

import pandas as pd

df = pd.read_csv(r"C:\Users\fpasha\OneDrive - Red River College Polytech\COMP-2040\final_project\data\Manitoba_Rabies_Surveillance_Data_Table.csv")

df.head()

,Region,Fiscal_Year,Date,Year,Species,Result,OBJECTID,ObjectId2
0,Winnipeg Regional Health Authority,2014-2015,2014/04/01 00:00:00+00,*2014,Cat,Negative,1,1
1,Southern Health-Santé Sud,2014-2015,2014/04/04 00:00:00+00,*2014,Cat,Negative,2,2
2,Interlake-Eastern Regional Health Authority,2014-2015,2014/04/09 00:00:00+00,*2014,Cat,Negative,3,3
3,Prairie Mountain Health,2014-2015,2014/04/14 00:00:00+00,*2014,Bovine,Negative,4,4
4,Southern Health-Santé Sud,2014-2015,2014/04/14 00:00:00+00,*2014,Dog,Negative,5,5


### 1.1 Dataset Shape

In [ ]:
# Quick check of dataset dimensions (rows, columns)
df.shape

(1306, 8)

### 1.2 Column Types and Missing Values


In [4]:
# Checking the structure of the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1306 entries, 0 to 1305
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Region       1305 non-null   object
 1   Fiscal_Year  1306 non-null   object
 2   Date         1306 non-null   object
 3   Year         1306 non-null   object
 4   Species      1306 non-null   object
 5   Result       1294 non-null   object
 6   OBJECTID     1306 non-null   int64 
 7   ObjectId2    1306 non-null   int64 
dtypes: int64(2), object(6)
memory usage: 81.8+ KB


#### 1.2.1 Missing Values Breakdown


In [10]:
# Direct count of missing values in each column
df.isna().sum()


Region          1
Fiscal_Year     0
Date            0
Year            0
Species         0
Result         12
OBJECTID        0
ObjectId2       0
dtype: int64

### 1.3 Descriptive Statistics


In [ ]:
# Getting a quick summary of all columns, including categorical ones.
# This helps me spot weird values, formatting issues, or categories that need cleaning.
df.describe(include='all')

,Region,Fiscal_Year,Date,Year,Species,Result,OBJECTID,ObjectId2
count,1305,1306,1306,1306,1306,1294,1306.000000,1306.000000
unique,7,11,982,11,9,3,NaN,NaN
top,Prairie Mountain Health,2018-2019,2017/09/29 00:00:00+00,*2018,Dog,Negative,NaN,NaN
freq,408,149,6,153,381,1107,NaN,NaN
mean,NaN,NaN,NaN,NaN,NaN,NaN,653.500000,653.500000
std,NaN,NaN,NaN,NaN,NaN,NaN,377.154036,377.154036
min,NaN,NaN,NaN,NaN,NaN,NaN,1.000000,1.000000
25%,NaN,NaN,NaN,NaN,NaN,NaN,327.250000,327.250000
50%,NaN,NaN,NaN,NaN,NaN,NaN,653.500000,653.500000
75%,NaN,NaN,NaN,NaN,NaN,NaN,979.750000,979.750000


### 1.4 Initial Observations

1. The dataset has 1,306 rows and 8 columns, covering multiple fiscal years of rabies surveillance in Manitoba.
2. The `Date` column is stored as a string with timezone formatting (`+00`), so it will need to be converted into a proper datetime format.
3. The `Year` column contains values like `*2018`, which means there are formatting issues that need to be cleaned.
4. The `Result` column has three main categories (Positive, Negative, Unfit), but also contains some missing values.
5. The `Species` column has 9 unique species, with dogs being the most common submission.
6. The `Region` column has one missing value and seven unique regions, with Prairie Mountain Health being the most frequent.


### 1.5 Analytical Questions

1. Which species are most commonly submitted for rabies testing across Manitoba?
2. How do positive rabies cases change over time (by year or fiscal year)?
3. Which regions submit the most samples, and do certain regions have higher positivity rates?
